src/interfaces/ecpg/ecpglib/misc.c

```diff
@@ -303,7 +303,7 @@ ecpg_log(const char *format,...)
 	/* dump out internal sqlca variables */
 	if (ecpg_internal_regression_mode && sqlca != NULL)
 	{
-		fprintf(debugstream, "[NO_PID]: sqlca: code: %ld, state: %s\n",
+		fprintf(debugstream, "[NO_PID]: sqlca: code: %ld, state: %5s\n",
```

Comment:
Can the `sqlstate` be a string with less than 5 chars? In case it is, lets say, `abc`, the output will be padded with 2 spaces to `  abc`. Maybe better to change to "%.5s"? Then it will be a 'precision' specification, where for %s `Precision specifies the maximum number of bytes to be written`

Reply:
> Can the `sqlstate` be a string with less than 5 chars?

No, sqlstate always must have 5 characters

Reply:
Sample code below:
```
#include <stdio.h>

int main()
{
        char tmp[11] = "0123456789";
        printf("tmp is '%5s'\n", tmp);
        return 0;
}
```
outputs:
```
tmp is '0123456789'
```

while the code below:

```
#include <stdio.h>

int main()
{
        char tmp[11] = "0123456789";
        printf("tmp is '%.5s'\n", tmp);
        return 0;
}
```
outputs:
```
tmp is '01234'
```

thus to limit the output to 5 chars it is required to change the format.

Reply:
fixed